src/main/java/com/arpnetworking/metrics/mad/parsers/JsonToRecordParser.java

```diff
@@ -137,27 +142,18 @@ public Record parse(final byte[] data) throws ParsingException {
         putVariablesVersion2c(model.getCounters(), MetricType.COUNTER, variables);
         putVariablesVersion2c(model.getGauges(), MetricType.GAUGE, variables);
 
-        // NOTE: We are injecting context into annotations.
-        // The injected annotations will overwrite any specified by the user.
-        final ImmutableMap.Builder<String, String> mappedAnnotations = ImmutableMap.builder();
-        mappedAnnotations.put(Key.HOST_DIMENSION_KEY, _defaultHost);
-        mappedAnnotations.put(Key.SERVICE_DIMENSION_KEY, _defaultService);
-        mappedAnnotations.put(Key.CLUSTER_DIMENSION_KEY, _defaultCluster);
-        for (final Map.Entry<String, String> entry : annotations.getOtherAnnotations().entrySet()) {
-            final String key = entry.getKey();
-            final String value = entry.getValue();
-            if (!Key.HOST_DIMENSION_KEY.equals(key)
-                    && !Key.SERVICE_DIMENSION_KEY.endsWith(key)
-                    && !Key.CLUSTER_DIMENSION_KEY.endsWith(key)) {
-                mappedAnnotations.put(key, value);
-            }
-        }
+        // NOTE: We are injecting context into dimensions, values set in annotations will be ignored
+        final ImmutableMap.Builder<String, String> dimensionsBuilder = ImmutableMap.builder();
+        dimensionsBuilder.put(Key.HOST_DIMENSION_KEY, _defaultHost);
+        dimensionsBuilder.put(Key.SERVICE_DIMENSION_KEY, _defaultService);
+        dimensionsBuilder.put(Key.CLUSTER_DIMENSION_KEY, _defaultCluster);
 
         return new DefaultRecord.Builder()
                 .setMetrics(variables.build())
                 .setId(UUID.randomUUID().toString())
                 .setTime(timestamp)
-                .setAnnotations(mappedAnnotations.build())
+                .setAnnotations(ImmutableMap.copyOf(annotations.getOtherAnnotations()))
```

Comment:
This will now include the annotations excluded on previous lines 149-151 - is this intended?